src/test/java/seedu/address/model/internship/DurationTest.java

```diff
@@ -0,0 +1,64 @@
+package seedu.address.model.internship;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+class DurationTest {
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Duration(null));
+    }
+
+    @Test
+    public void constructor_invalidDuration_throwsIllegalArgumentException() {
+        String invalidDuration = "";
+        assertThrows(IllegalArgumentException.class, () -> new Duration(invalidDuration));
+    }
+
+    @Test
+    public void isValidDuration() {
+        // null duration
+        assertThrows(NullPointerException.class, () -> Duration.isValidDuration(null));
+
+        // invalid durations
+        assertFalse(Duration.isValidDuration("")); // empty string
+        assertFalse(Duration.isValidDuration(" ")); // spaces only
+        assertFalse(Duration.isValidDuration("a")); // non number
+        assertFalse(Duration.isValidDuration("2.2")); // non-integer number
+        assertFalse(Duration.isValidDuration("-2")); // non-positive number
+
+        // valid durations
+        assertTrue(Duration.isValidDuration("1"));
+        assertTrue(Duration.isValidDuration("2"));
+        assertTrue(Duration.isValidDuration("3"));
+        assertTrue(Duration.isValidDuration("4"));
+        assertTrue(Duration.isValidDuration("5"));
+        assertTrue(Duration.isValidDuration("6"));
+        assertTrue(Duration.isValidDuration("7"));
+        assertTrue(Duration.isValidDuration("8"));
+        assertTrue(Duration.isValidDuration("9"));
+    }
```

Comment:
Good effort but I think it is a bit redundant. Maybe you should test very large integers instead?